components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/IdentityProvider.java

```diff
@@ -365,19 +384,161 @@ public void setAlias(String alias) {
     }
 
     /**
-     * @return
+     * Get certificate
+     * @return if certificate is in JSON array then return only first element.
      */
     public String getCertificate() {
+
+        // Check whether certificate is in json format.
+        if (StringUtils.isNotBlank(certificate) && certificate.startsWith(JSON_ARRAY_IDENTIFIER)) {
+            if (!certificate.equals(EMPTY_JSON_ARRAY)) {
+                certificate = ((JSONObject) (new JSONArray(certificate).get(0))).getString(CERT_VALUE);
+            } else {
+                // If certificate is an empty json array, then return empty value.
+                certificate = "";
+            }
+        }
         return certificate;
     }
 
     /**
      * @param certificate
      */
     public void setCertificate(String certificate) {
+
+        setCertificateInfoArray(certificate);
         this.certificate = certificate;
     }
 
+    /**
+     * Get certificate info array
+     * @return CertificateInfo array
+     */
+    public CertificateInfo[] getCertificateInfoArray() {
+
+        return certificateInfoArray;
+    }
+
+    /**
+     * Set certificateInfoArray
+     *
+     * @param certificateValue array which contains certificate info. Here Certificate info contains thumbPrint
+     *                             and certificate value as attributes.
+     */
+    private void setCertificateInfoArray(String certificateValue) {
+
+        try {
+            if (StringUtils.isNotBlank(certificateValue) && !certificateValue.equals(EMPTY_JSON_ARRAY)) {
+                certificateValue = certificateValue.trim();
+                try {
+                    this.certificateInfoArray = handleJsonFormatCertificate(certificateValue);
+                } catch (JSONException e) {
+                    // Handle plain text certificate for file based configuration.
+                    if (certificateValue.startsWith(IdentityUtil.PEM_BEGIN_CERTFICATE)) {
+                        this.certificateInfoArray = handlePlainTextCertificate(certificateValue);
+                    } else {
+                        // Handle encoded certificate values. While uploading through UI.
+                        this.certificateInfoArray = handleEncodedCertificate(certificateValue);
+                    }
+                }
+            }
+        } catch (NoSuchAlgorithmException e) {
+            log.error("Error while generating thumbPrint. Unsupported hash algorithm. " + e);
```

Comment:
The stacktrace is not logged here

Reply:
it's handle inside the private method createCertificateInfo().